src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -57,121 +57,42 @@ trait ChoosesMode {
-        case ChoosesModeData(
-            BasePersonData(_, _, _, _, _, Some(vehicle), _, _, _, _, _, _),
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _
-            ) =>
-          self ! MobilityStatusResponse(Vector(beamVehicles(vehicle)))
+        case modeChoiceData: ChoosesModeData if modeChoiceData.personData.currentTourPersonalVehicle.isDefined =>
+          self ! MobilityStatusResponse(Vector(beamVehicles(modeChoiceData.personData.currentTourPersonalVehicle.get)))
         // Only need to get available street vehicles if our mode requires such a vehicle
-        case ChoosesModeData(
-            BasePersonData(
-              currentActivityIndex,
-              _,
-              _,
-              _,
-              None | Some(CAR | BIKE | DRIVE_TRANSIT),
-              _,
-              _,
-              _,
-              _,
-              _,
-              _,
-              _
-            ),
-            currentLocation,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _,
-            _
-            ) =>
+        case modeChoiceData: ChoosesModeData
+            if modeChoiceData.personData.currentTourMode.isEmpty ||
+            modeChoiceData.personData.currentTourMode.contains(CAR) ||
+            modeChoiceData.personData.currentTourMode.contains(BIKE) ||
+            modeChoiceData.personData.currentTourMode.contains(DRIVE_TRANSIT) =>
           implicit val executionContext: ExecutionContext = context.system.dispatcher
           Future
             .sequence(
               vehicleFleets.map(
                 _ ? MobilityStatusInquiry(
                   id,
-                  currentLocation,
-                  _experiencedBeamPlan
-                    .activities(currentActivityIndex)
+                  modeChoiceData.currentLocation,
+                  _experiencedBeamPlan.activities(modeChoiceData.personData.currentActivityIndex)
                 )
               )
             )
-            .map(
-              listOfResponses =>
-                MobilityStatusResponse(
-                  listOfResponses
-                    .collect {
-                      case MobilityStatusResponse(vehicles) =>
-                        vehicles
-                    }
-                    .flatten
-                    .toVector
+            .map { listOfResponses =>
+              MobilityStatusResponse(
+                listOfResponses.flatMap { case MobilityStatusResponse(vehicles) => vehicles }.toVector
```

Comment:
I don't think this is equal transformation. Imagine you have the following code:
```scala
val seq: Seq[Any] = List[Any]("Hello", 1, 4.0)
```
Then if we use collect, it will work as expected:
```scala
seq.collect { case s: String => s }
```

But if I use map/flatMap, it will throw `scala.MatchError: 1 (of class java.lang.Integer)`:
```scala
seq.map { case s: String => s }
```
![image](https://user-images.githubusercontent.com/5107562/91856176-64def300-ec90-11ea-9665-c7124d2c0132.png)

Reply:
fixed